src/thunderbots/software/ai/hl/stp/stp.cpp

```diff
@@ -8,28 +8,59 @@
 #include <random>
 
 #include "ai/hl/stp/play/play.h"
+#include "ai/hl/stp/play/stop_play.h"
 #include "ai/hl/stp/tactic/tactic.h"
-#include "ai/intent/move_intent.h"
+#include "ai/intent/stop_intent.h"
 #include "util/logger/init.h"
+#include "util/parameter/dynamic_parameters.h"
 
 STP::STP(long random_seed) : random_number_generator(random_seed) {}
 
 std::vector<std::unique_ptr<Intent>> STP::getIntents(const World& world)
 {
+    bool override_play = Util::DynamicParameters::AI::override_ai_play.value();
+    bool override_play_value_changed =
+        Util::DynamicParameters::AI::override_ai_play.valueChanged();
+    std::string override_play_name = Util::DynamicParameters::AI::current_ai_play.value();
+    bool override_play_name_value_changed =
+        Util::DynamicParameters::AI::current_ai_play.valueChanged();
+    auto all_play_names = PlayFactory::getRegisteredPlayNames();
+
     // Assign a new play if we don't currently have a play assigned, the current play's
     // invariant no longer holds, or the current play is done
-    if (!current_play || !current_play->invariantHolds(world) || current_play->done())
+    if (!current_play || !current_play->invariantHolds(world) || current_play->done() ||
+        override_play_name_value_changed || override_play_value_changed)
```

Comment:
shouldn't the `override` override this?

Reply:
No. These are conditions for if the play should change. The play could change if it's no longer valid, OR if we change an override parameter (otherwise we'd override a play but it wouldn't change until the current play is done, which is not the desired behavior)